### PR TITLE
UF-8502: Extended service to handle sending of sms batches

### PR DESCRIPTION
**Pull request by josc04**
- Added new endpoint to call for sending sms batches
- Extended service to handle sms batches
- Added IT-tests for sms batches
- Removed unused imports and cleaned up some redundant code
- Added unit tests for some classes and methods that were missing unit tests

### CODE REVIEW

File: src/main/java/se/sundsvall/messaging/service/MessageEventDispatcher.java [hunk 1]
```diff
@@ -72,6 +88,23 @@ public InternalDeliveryResult handleSmsRequest(final SmsRequest request) {
...
+	public InternalDeliveryBatchResult handleSmsBatchRequest(final SmsBatchRequest request) {
+		var batchId = UUID.randomUUID().toString();
+
+		var deliveryResults = ofNullable(request.parties()).orElse(emptyList()).stream()
+			.filter(party -> isWhitelisted(SMS, party.mobileNumber()))
+			.map(party -> requestMapper.toSmsRequest(request, party))
+			.map(message -> messageMapper.toMessage(message, batchId))
+			.map(dbIntegration::saveMessage)
+			.map(message -> publishMessageEvent(message.withBatchId(batchId)))
+			.toList();
```
Comment (AndreSoderlund) [inline src/main/java/se/sundsvall/messaging/service/MessageEventDispatcher.java:100]: Could possibly add a mapping method that maps the request, part  and batch id directly to a message.

File: src/main/java/se/sundsvall/messaging/service/MessageEventDispatcher.java [hunk 2]
```diff
@@ -72,6 +88,23 @@ public InternalDeliveryResult handleSmsRequest(final SmsRequest request) {
...
 
+	public InternalDeliveryBatchResult handleSmsBatchRequest(final SmsBatchRequest request) {
+		var batchId = UUID.randomUUID().toString();
+
+		var deliveryResults = ofNullable(request.parties()).orElse(emptyList()).stream()
+			.filter(party -> isWhitelisted(SMS, party.mobileNumber()))
+			.map(party -> requestMapper.toSmsRequest(request, party))
+			.map(message -> messageMapper.toMessage(message, batchId))
+			.map(dbIntegration::saveMessage)
+			.map(message -> publishMessageEvent(message.withBatchId(batchId)))
```
Comment (cheezi0747) [inline src/main/java/se/sundsvall/messaging/service/MessageEventDispatcher.java:99]: Is `batchId` already set with the call to `messageMapper.toMessage(message, batchId)` ?

File: src/test/java/se/sundsvall/messaging/service/MessageEventDispatcherTests.java [hunk 3]
```diff
@@ -1,47 +1,63 @@
...
     @Mock
     private BlacklistService mockBlacklistService;
+
     @Mock
     private DbIntegration mockDbIntegration;
 
     @Mock
     private MessageMapper mockMessageMapper;
 
+	@Mock
```
Comment (cheezi0747) [inline src/test/java/se/sundsvall/messaging/service/MessageEventDispatcherTests.java:58]: Indentation looks off

File: src/main/java/se/sundsvall/messaging/service/MessageEventDispatcher.java (hunk 1 above)
Comment (josc04) [reply src/main/java/se/sundsvall/messaging/service/MessageEventDispatcher.java:100]: Even better - row 97 already maps the batchId into the message object, so I'll change this row to be:
```
.map(this::publishMessageEvent)
``` 
instead.

**Review by AndreSoderlund — APPROVED** [review]

**Pull request merged**